docs/src/LM.md

```diff
@@ -0,0 +1,180 @@
+
+**TextAnalysis** provide following different Language Models 
+
+- **MLE** - Base Ngram model.
+- **Lidstone **- Base Ngram model with Lidstone smoothing.
+- **Laplace** - Base Ngram language model with Laplace smoothing.
+- **WittenBellInterpolated** - Interpolated Version of witten-Bell algorithm.
+- **KneserNeyInterpolated** - Interpolated  version of Kneser -Ney smoothing.
+
+## APIs
+
+To use the API, we first *Instantiate* desired model and then load it with train set
+
+```julia
+<MLE>(word::Vector{T}, unk_cutoff=1, unk_label="<unk>") where { T <: AbstractString}
+        
+Lidstone(word::Vector{T}, gamma:: Float64, unk_cutoff=1, unk_label="<unk>") where { T <: AbstractString}
+        
+Laplace(word::Vector{T}, unk_cutoff=1, unk_label="<unk>") where { T <: AbstractString}
+        
+WittenBellInterpolated(word::Vector{T}, unk_cutoff=1, unk_label="<unk>") where { T <: AbstractString}
+        
+KneserNeyInterpolated(word::Vector{T}, discount:: Float64=0.1, unk_cutoff=1, unk_label="<unk>") where { T <: AbstractString}
+        
+(lm::<Languagemodel>)(text, min::Integer, max::Integer)
+```
+Arguments:
+
+ * `word` : Array of  strings to store vocabulary.
+
+ * `unk_cutoff`: Tokens with counts greater than or equal to the cutoff value will be considered part of the vocabulary.
+
+ * `unk_label`: token for unkown labels 
+
+ *  `gamma`: smoothing arugment gamma 
+
+ * `discount`:  discouting factor for `KneserNeyInterpolated`
+
+   for more information see docstrings of vocabulary
+
+```julia
+julia> voc = ["my","name","is","salman","khan","and","he","is","shahrukh","Khan"]
+
+julia>train = ["khan","is","my","good", "friend","and","He","is","my","brother"]
+# voc and train are used to train vocabulary and model respectively
+
+julia> model = MLE(voc)
+MLE(Vocabulary(Dict("khan"=>1,"name"=>1,"<unk>"=>1,"salman"=>1,"is"=>2,"Khan"=>1,"my"=>1,"he"=>1,"shahrukh"=>1,"and"=>1…), 1, "<unk
+        >", ["my", "name", "is", "salman", "khan", "and", "he", "is", "shahrukh", "Khan", "<unk>"]))
+julia> print(voc)
+11-element Array{String,1}:
+ "my"      
+ "name"    
+ "is"      
+ "salman"  
+ "khan"    
+ "and"     
+ "he"      
+ "is"      
+ "shahrukh"
+ "Khan"    
+ "<unk>"   
+# you can see "<unk>" token is added to voc 
+julia> fit = model(train,2,2) #considering only bigrams
+julia> unmaskedscore = score(model, fit, "is" ,"<unk>") #score output P(word | context) without replacing context word with "<unk>"
+0.3333333333333333
+julia> masked_score = maskedscore(model,fit,"is","alien")
+0.3333333333333333
+#as expected maskedscore is equivalent to unmaskedscore with context replaced with "<unk>"
+
+```
+!!! note
+
+    When you call `MLE(voc)` for the first time, It will update your vocabulary set as well. 
+
+## Evaluation Method
+
+- `score` 
+
+    used to evaluate probablity of word given context (*P(word | context)*)
+
+   ```julia
+	score(m::gammamodel, temp_lm::DefaultDict, word::AbstractString, context::AbstractString)
```

Comment:
Check this locally once, but I don't think this will be rendered as expected from markdown to HTML format due to indentation of ```julia.
If you want to keep - score as sub-section under Evaluation Method. Then try the following instead.

```suggestion
### `score` 

    used to evaluate the probability of word given context (*P(word | context)*)

```julia
	score(m::gammamodel, temp_lm::DefaultDict, word::AbstractString, context::AbstractString)
```
If you go this way then, incorporate similar changes in the next 4 evaluation measures.

Reply:
> I don't think this will be rendered as expected from markdown to HTML format.

I am not sure about this, do test this locally once.